Add @znewman01 to reusable-release

znewman01 opened the pull request:
I am a Cosign owner and occasionally need to cut a Cosign release.

Signed-off-by: Zachary Newman <zjn@chainguard.dev>

<!--
Thanks for opening a pull request! Please do not just delete this text.  The three fields below are mandatory.

Please remember to:
- This PR requires an issue. If it is a new feature, the issue should proceed the PR and will have allowed sufficent time for discussions to take place. Pleases use
issue tags such as "Closes #XYZ" or "Resolves sigstore/repo-name#XYZ".
  [documentation](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
- ensure your commits are signed-off, as sigstore uses the [DCO](https://en.wikipedia.org/wiki/Developer_Certificate_of_Origin) using `git commit -s`, or `git commit -s --amend` if you want to amend already existing commits
- lastly, ensure there are no merge commits!

Thank you :)
-->

#### Summary
<!--
 Explain the **motivation** for making this change. What existing problem does the pull request solve? How can reviewers test this PR?
-->

#### Release Note
<!--
Add a release note for each of the following conditions:

* Config changes (additions, deletions, updates)
* API additions—new endpoint, new response fields, or newly accepted request parameters
* Database changes (any)
* Websocket additions or changes
* Anything noteworthy to an administrator running private sigstore instances (err on the side of over-communicating)
* New features and improvements, including behavioural changes, UI changes and CLI changes
* Bug fixes and fixes of previous known issues
* Deprecation warnings, breaking changes, or compatibility notes

If no release notes are required write NONE. Use past-tense.

-->

#### Documentation
<!--

Does this change require an update to documentation? How will users implement your new feature?

Please reference a PR within https://docs.sigstore.dev

-->

haydentherapper reviewed (approved):
We should probably move these reusable workflows to a dedicated project at some point. 

Also Lgtm, but we need to rethink who should be able to cut releases, since this isn’t sustainable. We have an issue in public good instance if anyone wants to chime in!

haydentherapper commented on .github/workflows/reusable-release.yml:
```diff
@@ -42,7 +42,7 @@ jobs:
     steps:
       - name: Check actor access
-        if: ${{ !contains( fromJson('["bobcallaway","cpanato","dlorenc","lukehinds","priyawadhwa","vaikas","haydentherapper","afeddersen"]'), github.actor ) }}
+        if: ${{ !contains( fromJson('["bobcallaway","cpanato","dlorenc","lukehinds","priyawadhwa","vaikas","haydentherapper","afeddersen","znewman01"]'), github.actor ) }}
```
Could we remove afeddersen?

cpanato commented:
working to remove this and use github actions only and not more cloudbuild, will try to get this in the xmas time ready

znewman01 commented:
Addressed comments so ready for re-approval.

+1 to the comments about sustainability here, thanks @cpanato you're doing incredible work as usual 🎉
